src/ASM/ASMs1D.C

```diff
@@ -1482,52 +1583,60 @@ bool ASMs1D::assembleL2matrices (SparseMatrix& A, StdVector& B,
   Vector phi(p1);
   Matrix dNdu, Xnod, J;
 
+  const IntVec& mlge = projMLGE.empty() ? MLGE : projMLGE;
+  const IntMat& mnpc = projMNPC.empty() ? MNPC : projMNPC;
+
 
   // === Assembly loop over all elements in the patch ==========================
 
   int ip = 0;
-  for (size_t iel = 0; iel < nel; iel++)
+  for (size_t iel = 0; iel < mlge.size(); iel++)
   {
-    if (MLGE[iel] < 1) continue; // zero-area element
+    if (mlge[iel] < 1) continue; // zero-area element
 
     if (continuous)
     {
       // Set up control point (nodal) coordinates for current element
-      if (!this->getElementCoordinates(Xnod,1+iel))
+      if (!this->getElementCoordinates(Xnod,1+iel,mnpc,proj))
         return false;
-      else if ((dL = this->getParametricLength(1+iel)) < 0.0)
+
+      int inod1 = mnpc[iel][proj->order()-1];
+      dL = *(proj->basis().begin()+inod1+1) -
+           *(proj->basis().begin()+inod1);
+      if (dL < 0.0)
```

Comment:
i guess we want to skip for zero knotspans?

Reply:
yes, but that (I think) is already been taken care of by checking `if (mlge[iel] < 1)` earlier on

Reply:
right.